app/src/main/java/org/fossasia/phimpme/shareActivity.java

```diff
@@ -21,109 +29,184 @@
+    String caption;
+    boolean atleastOneShare = false;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_share);
-        mEditText = (EditText) findViewById(R.id.edit_text);
-        sendMessage = mEditText.getText().toString();
         setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+        setupUI();
         initView();
-        buttonClick();
     }
 
-    public void buttonClick() {
-        mshareButton = (ImageView) findViewById(R.id.share_button);
-        mshareButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                shareButton();
-            }
-        });
-        mfacebookButton = (ImageView) findViewById(R.id.facebook_button);
-        mfacebookButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                shareFacebook();
-            }
-        });
-        mInstagrambutton = (ImageView) findViewById(R.id.instagram_button);
-        mInstagrambutton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                shareInstagram();
-            }
-        });
-        fab_home = (FloatingActionButton) findViewById(R.id.fab_home);
-        fab_home.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                goToHome();
-            }
-        });
-    }
+    private void setupUI() {
 
-    private void shareButton() {
-        sendMessage = mEditText.getText().toString();
-        Uri uri = Uri.fromFile(new File(saveFilePath));
-        Intent shareIntent = new Intent();
-        shareIntent.setAction(Intent.ACTION_SEND);
-        shareIntent.putExtra(Intent.EXTRA_TEXT, sendMessage);
-        shareIntent.putExtra(Intent.EXTRA_STREAM, uri);
-        shareIntent.setType("image/jpeg");
-        startActivity(Intent.createChooser(shareIntent, getResources().getText(R.string.send_image)));
+        toolbar = (Toolbar) findViewById(R.id.toolbar);
+        toolbar.setTitleTextColor(Color.BLACK);
+        toolbar.setTitle(R.string.shareto);
+        toolbar.setBackgroundColor(Color.WHITE);
+        Drawable backIcon = getToolbarIcon(GoogleMaterial.Icon.gmd_arrow_back);
+        backIcon.setColorFilter(Color.BLACK, PorterDuff.Mode.SRC_ATOP);
+        toolbar.setNavigationIcon(backIcon);
+        setSupportActionBar(toolbar);
+
+        setStatusBarColor(getResources().getColor(R.color.md_grey_400));
+
+        parent = findViewById(R.id.share_layout);
+        shareImage = (ImageView)findViewById(R.id.share_image);
+
+        text_caption = (EditText)findViewById(R.id.edittext_share_caption);
+
+        cell00 = findViewById(R.id.cell_00);
```

Comment:
@vinaysajjanapu We are already using butterknife in the project, so use that instead of conventional finviewbyid calls.